salt/proxy/junos.py

```diff
@@ -101,6 +110,23 @@ def init(opts):
     if 'username' in opts['proxy'].keys():
         opts['proxy']['user'] = opts['proxy'].pop('username')
     proxy_keys = opts['proxy'].keys()
+
+    # If encoded_passwd is found, prefer it over passwd or password
+    if 'encoded_password' in opts['proxy'].keys() and HAS_JUNOSSECURE:
```

Comment:
One optimisation we can do. We can avoid calling opts['proxy'].keys() two times. But I am not so much worried about it because this code path is very least called. See if you can do it otherwise leave it.
```
if 'encoded_password' in opts['proxy'].keys():
    if HAS_JUNOSSECURE:
          code for decode_password
    else:
       opts['proxy'].pop('encoded_password')
        log.error('encoded_password option provided, but could not find junossecure option to decode.'
                  ' Proceeding with passwd or password options if provided.')
```

Reply:
Thanks for pointing it out. Changed the logic.